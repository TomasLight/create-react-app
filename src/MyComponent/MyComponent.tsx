import React from 'react';

interface IMyComponentProps {
    text: string;
}

type Props = IMyComponentProps;

const MyComponent = (props: Props) => {
  const { text } = props;

  return (
    <p style={{ color: 'red' }}>
      {text}
    </p>
  );
};

export { MyComponent };
