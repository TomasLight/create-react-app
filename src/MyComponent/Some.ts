export class Some {
    myProperty: string;

    constructor() {
        this.myProperty = '';
    }
}
