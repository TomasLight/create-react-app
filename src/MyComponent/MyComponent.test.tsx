import React from "react";
import { render } from '@testing-library/react';
import { MyComponent } from './MyComponent';

it("check render", () => {
    const text = "some text";
    const component = (
        <MyComponent text={text} />
    );

    const { container, getByText } = render(component);
    expect(container).toMatchSnapshot();
    expect(getByText(text)).toBeVisible();
});
