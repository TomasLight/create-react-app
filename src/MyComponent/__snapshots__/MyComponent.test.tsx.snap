// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render 1`] = `
<div>
  <p
    style="color: red;"
  >
    some text
  </p>
</div>
`;
